task/monorail-server/src/bsp/sidecar_bcd.rs

```diff
@@ -548,6 +585,31 @@ impl<'a, R: Vsc7448Rw> Bsp<'a, R> {
         let phy = vsc85xx::Phy::new(phy_port, &mut phy_rw);
         Some(callback(phy))
     }
+
+    pub(crate) fn unlock_vlans_until(
+        &mut self,
+        unlock_until: u64,
+    ) -> Result<(), MonorailError> {
+        ringbuf_entry!(Trace::UnlockUntil(unlock_until));
+        self.vsc7448
+            .configure_vlan_sidecar_unlocked()
+            .map_err(|e| {
+                ringbuf_entry!(Trace::UnlockError(e));
+                MonorailError::from(e)
+            })?;
+        self.vlan_mode = VLanMode::UnlockedUntil(unlock_until);
```

Comment:
Nitpicky question, but - is it possible for `configure_vlan_sidecar_unlocked()` to partially succeed and leave us in some bad state while `self.vlan_mode` still thinks we're in `VLanMode::Locked`?